src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/talismans/Talisman.java

```diff
@@ -130,48 +134,71 @@ private static boolean hasMessage(Talisman talisman) {
         return talisman.getMessageSuffix() != null;
     }
 
-    public static boolean checkFor(Event e, SlimefunItemStack stack) {
-        return checkFor(e, stack.getItem());
+    public static boolean tryActivate(@Nonnull Event e, @Nonnull SlimefunItemStack stack) {
+        return (tryActivateAndGet(e, stack.getItem()) != null);
+    }
+
+    public static boolean tryActivate(@Nonnull Event e, @Nonnull SlimefunItem item) {
+        return (tryActivateAndGet(e, item) != null);
     }
 
-    public static boolean checkFor(Event e, SlimefunItem item) {
+    public static ItemStack tryActivateAndGet(@Nonnull Event e, @Nonnull SlimefunItemStack stack) {
+        return tryActivateAndGet(e, stack.getItem());
+    }
+
+    @Nullable
+    public static ItemStack tryActivateAndGet(@Nonnull Event e, @Nonnull SlimefunItem item) {
         if (!(item instanceof Talisman)) {
-            return false;
+            return null;
         }
 
         Talisman talisman = (Talisman) item;
         if (ThreadLocalRandom.current().nextInt(100) > talisman.getChance()) {
-            return false;
+            return null;
         }
 
         Player p = getPlayerByEventType(e);
         if (p == null || !pass(p, talisman)) {
-            return false;
+            return null;
+        }
+
+        if (!Slimefun.hasUnlocked(p, talisman, true)) {
+            return null;
         }
```

Comment:
This will send a message to the player, we don't want this if the Player isn't even carrying the talisman.